grammar.js

```diff
@@ -613,7 +614,10 @@ module.exports = grammar({
         alias($._pattern_binary_expression, $.binary_expression)
       ),
     _pattern_binary_expression: ($) =>
-      binaryExpr(prec.left, 1, "<>", $._pattern_expression),
+      choice(
+        binaryExpr(prec.left, 1, "<>", $._pattern_expression),
+        binaryExpr(prec.left, 1, "as", $._pattern_expression)
```

Comment:
The right hand side is a variable rather than a pattern (assuming I'm reading this correctly)

Reply:
Ah yep, we can restrict this so that the left-hand side is always a string literal and the right-hand side is always an identifier